src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -56,13 +73,37 @@ class InstallAppsTask extends AbstractTask {
             throw new GradleException("The project archive was not found and cannot be installed.")
         }
         Files.copy(archive.toPath(), new File(getServerDir(project), "/" + project.liberty.installapps.appsDirectory + "/" + getArchiveName(archive.getName())).toPath(), StandardCopyOption.REPLACE_EXISTING)
+    
+        validateAppConfig(getArchiveName(archive.getName()), getBaseName(archive))
     }
     
-    private String getInstallAppPackages() {
-        if (project.plugins.hasPlugin("ear")) {
-            project.liberty.installapps.installAppPackages = "project"
+    private void validateAppConfig(String fileName, String artifactId) throws Exception {
+        String appsDir = project.liberty.installapps.appsDirectory
+        
+        if(appsDir.equalsIgnoreCase('apps') && !isAppConfiguredInSourceServerXml(fileName)){
+            applicationXml.createApplicationElement(fileName, artifactId)
+        }
+        else if(appsDir.equalsIgnoreCase('dropins') && isAppConfiguredInSourceServerXml(fileName)){
+            throw new GradleException("The application is configured in the server.xml and the plug-in is configured to install the application in the dropins folder. A configured application must be installed to the apps folder.")
         }
-        return project.liberty.installapps.installAppPackages
+    }
+    
+    protected boolean isAppConfiguredInSourceServerXml(String fileName) {
+        boolean configured = false; 
+        if (project.liberty.configFile != null && project.liberty.configFile.exists()) {
```

Comment:
What if we are using configDirectory that includes a server.xml file, does configFile get set?
I couldn't find where it gets set.